README.md

```diff
@@ -1 +1,510 @@
 # react-hook-form-jsonschema
+
+> Small project based on [react-hook-form](https://github.com/react-hook-form/react-hook-form) that exposes an API for easily creating customizable forms based on a [JSON Schema](https://json-schema.org/understanding-json-schema/index.html) with built-in validation.
+
+## Table of Contents
+
+- [react-hook-form-jsonschema](#react-hook-form-jsonschema)
+  - [Simple Usage](#simple-usage)
+  - [Installation](#installation)
+  - [Components API](#components-api)
+    - [FormContext component](#formcontext-component)
+  - [Hooks API](#hooks-api)
+    - [useHidden(path)](#usehiddenpath)
+    - [useInput(path)](#useinputpath)
+    - [useObject(path, UISchema)](#useobjectpath-uischema)
+    - [usePassword(path)](#usepasswordpath)
+    - [useRadio(path)](#useradiopath)
+    - [useSelect(path)](#useselectpath)
+    - [useTextArea(path)](#usetextareapath)
+
+## Simple Usage
+
+Suppose you have a simple JSON Schema that stores a person's first name:
+
+```js
+const personSchema = {
+  $id: 'https://example.com/person.schema.json',
+  $schema: 'http://json-schema.org/draft-07/schema#',
+  title: 'Person',
+  type: 'object',
+  properties: {
+    firstName: {
+      type: 'string',
+      description: "The person's first name.",
+    },
+  },
+}
+```
+
+And suppose you want to create a form field for the firstName field, simply use the useInput() hook for this and then render the form using react
+
+```JSX
+function FirstNameField(props) {
+  const inputMethods = useInput('#/firstName');
+
+  return (
+    <FormContext schema={personSchema}>
+      <label {...inputMethods.getLabelProps()}>
+        {inputMethods.name}
+      </label>
+      <input {...inputMethods.getInputProps()} />
+    </FormContext>
+  )
+}
+```
+
+## Installation
+
+With npm:
+
+```
+npm install react-hook-form-jsonschema --save
+```
+
+Or with yarn:
+
+```
+yarn add react-hook-form-jsonschema
+```
+
+## Components API
+
+### FormContext component
+
+#### props:
+
+##### Optional:
+
+- `mode`: String to indicate when to validate the input, default is `'onSubmit'`.
```

Comment:
I decided to keep the names that the react-hook-form lib already used, specially because I am simply forwarding this prop to the useForm hook defined in that library.

Reply:
Yeah, but it seems a bit strange since this lib e about the form and not about validation (maybe about both, but even then seems weird), may validationMode would be more declarative

Reply:
You make a good point, I will change the name in the next release